test/e2e/specs/editor/plugins/block-hooks.spec.js

```diff
@@ -0,0 +1,292 @@
+						expectedHookedBlockLastChild,
+					] );
+
+				const hookedBlock = editor.canvas.getByText(
+					getHookedBlockContent( 'last_child', blockType )
+				);
+				await editor.selectBlocks( hookedBlock );
+				await editor.clickBlockToolbarButton( 'Move up' );
+
+				// Save updated post.
+				const saveButton = page
+					.getByRole( 'region', { name: 'Editor top bar' } )
+					.getByRole( 'button', { name: 'Save', exact: true } );
+				await saveButton.click();
+				await page
+					.getByRole( 'button', { name: 'Dismiss this notice' } )
+					.filter( { hasText: 'updated' } )
+					.waitFor();
+
+				// Reload and verify that the new position of the hooked block has been persisted.
+				await page.reload();
+				await expect
+					.poll( editor.getBlocks )
+					.toMatchObject( [
+						{ name: 'core/heading' },
+						expectedHookedBlockAfterHeading,
+						expectedHookedBlockLastChild,
+						{ name: 'core/paragraph' },
+					] );
+
+				// Verify that the frontend reflects the changes made in the editor.
+				await page.goto( `/?p=${ containerPost.id }` );
+				await expect(
+					page.locator( '.entry-content > *' )
+				).toHaveClass( [
+					'wp-block-heading',
+					getHookedBlockClassName( 'after', 'core/heading' ),
+					getHookedBlockClassName( 'last_child', blockType ),
+					'dummy-paragraph',
+				] );
+			} );
+		} );
+	} );
+
+	test.describe( `Hooked blocks in Navigation Menu`, () => {
+		let postObject, containerPost;
+		test.beforeAll( async ( { requestUtils } ) => {
+			postObject = await requestUtils.createNavigationMenu( {
+				title: 'Navigation Menu',
+				status: 'publish',
+				content:
+					'<!-- wp:navigation-link {"label":"wordpress.org","url":"https://wordpress.org","kind":"custom"} /-->', // '<!-- wp:page-list /-->',
+			} );
+
+			await requestUtils.activatePlugin( 'gutenberg-test-block-hooks' );
+
+			// We need a container post to hold our block instance.
+			containerPost = await requestUtils.createPost( {
+				title: `Block Hooks in Navigation Menu`,
+				status: 'publish',
+				content: `<!-- wp:navigation {"ref":${ postObject.id }} /-->`,
+				meta: {
+					// Prevent Block Hooks from injecting blocks into the container
+					// post content so they won't distract from the ones injected
+					// into the block instance.
+					_wp_ignored_hooked_blocks: '["core/paragraph"]',
+				},
+			} );
+		} );
+
+		test.afterAll( async ( { requestUtils } ) => {
+			await requestUtils.deactivatePlugin( 'gutenberg-test-block-hooks' );
+
+			await requestUtils.deleteAllPosts();
+			await requestUtils.deleteAllBlocks();
+		} );
+
+		test( `should insert hooked blocks into Navigation Menu on frontend`, async ( {
+			page,
+		} ) => {
+			await page.goto( `/?p=${ containerPost.id }` );
+			await expect(
+				page.locator( '.wp-block-navigation__container > *' )
+			).toHaveClass( [
+				'wp-block-navigation-item wp-block-home-link',
+				' wp-block-navigation-item wp-block-navigation-link',
```

Comment:
Is this preceding space intentional?

Reply:
Yeah, the selector doesn't work without it 😕 I guess it's a (minor) bug in the block code.